tests/sps30-test.cpp

```diff
@@ -0,0 +1,155 @@
+                    "Mass concentration not rising monotonously");
+
+    // Check if number concentration is rising monotonously
+    CHECK_TRUE_TEXT(LEQ(SPS30_MIN_NC, m.nc_0p5) && LEQ(m.nc_0p5, m.nc_1p0) &&
+                        LEQ(m.nc_1p0, m.nc_2p5) && LEQ(m.nc_2p5, m.nc_4p0) &&
+                        LEQ(m.nc_4p0, m.nc_10p0) &&
+                        LEQ(m.nc_10p0, SPS30_MAX_NC),
+                    "Number concentration not rising monotonously");
+
+    // Set auto cleaning interval in days, reset and get it again
+    ret = sps30_set_fan_auto_cleaning_interval_days(set_interval_days);
+    CHECK_ZERO_TEXT(ret, "sps30_set_fan_auto_cleaning_interval_days");
+    sps30_delayed_reset();
+    ret = sps30_get_fan_auto_cleaning_interval_days(&get_interval_days);
+    CHECK_ZERO_TEXT(ret, "sps30_get_fan_auto_cleaning_interval_days");
+    CHECK_EQUAL_TEXT(set_interval_days, get_interval_days,
+                     "Fan auto cleaning interval days do not match");
+
+    // Set auto cleaning interval in days, reset and set it again
+    set_interval_seconds = set_interval_days * SECONDS_PER_DAY;
+    ret = sps30_set_fan_auto_cleaning_interval(set_interval_seconds);
+    CHECK_ZERO_TEXT(ret, "sps30_set_fan_auto_cleaning_interval");
+    sps30_delayed_reset();
+    ret = sps30_get_fan_auto_cleaning_interval(&get_interval_seconds);
+    CHECK_ZERO_TEXT(ret, "sps30_get_fan_auto_cleaning_interval");
+    CHECK_EQUAL_TEXT(set_interval_seconds, get_interval_seconds,
+                     "Fan auto cleaning interval seconds do not match");
+
+    // Start manual cleaning event
+    ret = sps30_start_manual_fan_cleaning();
+    CHECK_ZERO_TEXT(ret, "sps30_start_manual_fan_cleaning");
+
+    // Stop measurement
+    ret = sps30_stop_measurement();
+    CHECK_ZERO_TEXT(ret, "sps30_stop_measurement");
+}
+
+static void sps30_test_setup() {
+    uint8_t fs_major;
+    uint8_t fs_minor;
+    int16_t ret;
+    char serial[SPS30_MAX_SERIAL_LEN];
+
+    ret = sps30_probe();
+    CHECK_ZERO_TEXT(ret, "sps30_probe in test setup");
+
+    sps30_delayed_reset();
+
+    const char* version = sps_get_driver_version();
+    printf("sps30_get_driver_version: %s\n", version);
+
+    ret = sps30_get_serial(serial);
+    CHECK_ZERO_TEXT(ret, "sps30_get_serial");
+    printf("SPS30 serial: %s\n", serial);
+
+    ret = sps30_read_firmware_version(&fs_major, &fs_minor);
+    CHECK_ZERO_TEXT(ret, "sps30_read_firmware_version");
+    printf("SPS30 Firmware version: %u.%u\n", fs_major, fs_minor);
+}
+
+TEST_GROUP (SPSTestGroup) {
+    void setup() {
+        int16_t ret;
+        sensirion_i2c_init();
+        ret = sensirion_i2c_mux_set_single_channel(0x72, 3);
+        CHECK_EQUAL_TEXT(0, ret, "sensirion_i2c_mux_select_TEXT(0x72, 3)");
+        sps30_test_setup();
+    }
+
+    void teardown() {
+        sps30_delayed_reset();
+        sensirion_i2c_release();
+    }
+};
+
+TEST (SPSTestGroup, SPS30Test_no_cleaning) { sps30_test(0); }
```

Comment:
These look strange. Is it possible to configure clang-format to avoid this?

Reply:
Resolved offline: It is to cumbersome to get clang-format to behave correctly here. It seems to be a bug in clang-format. See https://stackoverflow.com/questions/51535549/format-foreach-macros-the-same-as-loops-with-clang-format